src/hooks/use-load-image.js

```diff
@@ -0,0 +1,27 @@
+import { usePromise } from "@nebula.js/stardust";
+import { getImageUrl } from "../utils/url-utils";
+
+// Create a promise that ensures the button background image is loaded properly
+// and the image url is resolved.
+const imagePromise = (url, app) =>
+  new Promise((resolve, reject) => {
+    const imgUrl = getImageUrl(url, app);
+    const img = new Image();
+    img.src = imgUrl;
+    img.onload = () => resolve(img);
+    img.onerror = () => reject(new Error(undefined));
+  });
+
+const useLoadImage = (layout, app) => {
+  const { url } = layout.style.background;
+  const [imageUrl] = usePromise(async () => {
+    if (url?.qStaticContentUrl?.qUrl) {
+      const imgSrc = await imagePromise(url.qStaticContentUrl.qUrl, app);
+      return imgSrc.src;
+    }
+    return null;
+  }, [layout]);
```

Comment:
`layout` is not being used inside the hook. So no point any "reloading" the image every time the layout changes.

Also not sure why you don't get a lint error about the missing `app` dependency.

```suggestion
  }, [url, app]);
```

Reply:
I used layout at line 16 and is used in `imagePromise`. I fixed the missing app.

Reply:
Yes but that is outside of the `usePromise` hook?

Reply:
I'm not sure how the hook counts it but seems putting url affects result the same. I'll fix it.